### PR TITLE
WIP: refactor: Introduce - fixes improper usage of thread

### PR DESCRIPTION
**Pull request by soluchok**
* TODO

Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: pkg/didcomm/protocol/introduce/models.go [hunk 1]
```diff
@@ -7,18 +7,20 @@ SPDX-License-Identifier: Apache-2.0
...
 
 // Proposal defines proposal request
 type Proposal struct {
-	Type   string            `json:"@type,omitempty"`
-	ID     string            `json:"@id,omitempty"`
-	To     *To               `json:"to,omitempty"`
-	NWise  bool              `json:"nwise,omitempty"`
-	Thread *decorator.Thread `json:"~thread,omitempty"`
-	Timing *decorator.Timing `json:"~timing,omitempty"`
+	service.Metadata `json:",squash"`  // nolint: staticcheck
```
Comment (llorllale) [inline pkg/didcomm/protocol/introduce/models.go]: The protocol's models should only have properties that travel over the wire.

File: pkg/client/introduce/client.go [hunk 2]
```diff
@@ -51,176 +38,63 @@ func New(ctx Provider, inv *didexchange.Invitation) (*Client, error) {
...
-//  - HandleRequestWithInvitation
-// usage: e.Continue(c.InvitationEnvelope(threadID))
-func (c *Client) InvitationEnvelope(thID string) *InvitationEnvelope {
-	opts, err := c.getInvitationEnvelope(thID)
-
-	if errors.Is(err, storage.ErrDataNotFound) {
-		return &InvitationEnvelope{Inv: c.defaultInv}
-	}
+// SendProposal sends a proposal to the introducees (the client does not have a public Invitation).
+func (c *Client) SendProposal(recipient1, recipient2 *introduce.Recipient) error {
```
Comment (llorllale) [inline pkg/client/introduce/client.go]: The target `To` should be kept separate from the recipients.

File: pkg/didcomm/protocol/introduce/models.go (hunk 1 above)
Comment (soluchok) [reply pkg/didcomm/protocol/introduce/models.go]: Done!

File: pkg/client/introduce/client.go (hunk 2 above)
Comment (llorllale) [reply pkg/client/introduce/client.go]: Disregard - proposal for recipient 1 is sent to recipient 2 and vice versa.

**Comment by troyronda** [comment]
@soluchok is this PR replaced by the other ones you opened?

**Comment by soluchok** [comment]
> @soluchok is this PR replaced by the other ones you opened?

Yes, it is.

**Pull request closed without merging**